File: app/src/main/java/net/feheren_fekete/applist/launcher/model/PageData.java
package net.feheren_fekete.applist.launcher.model;

public class PageData {

    public static final int TYPE_APPLIST_PAGE = 1;
    public static final int TYPE_WIDGET_PAGE = 2;

    private long mId;
    private int mType;
    private boolean mIsMainPage;

    public PageData(long id, int type, boolean isMainPage) {
        mId = id;
        mType = type;
        mIsMainPage = isMainPage;
    }

    public PageData(PageData other) {
        this(other.mId, other.mType, other.mIsMainPage);
    }

    public long getId() {
        return mId;
    }

    public int getType() {
        return mType;
    }

    public boolean isMainPage() {
        return mIsMainPage;
    }

    public void setMainPage(boolean mainPage) {
        mIsMainPage = mainPage;
    }

}

File: app/src/main/java/net/feheren_fekete/applist/launcher/model/PageData.kt
package net.feheren_fekete.applist.launcher.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class PageData(@PrimaryKey(autoGenerate = true) val id: Long,
                    val type: Int,
                    val isMainPage: Boolean,
                    val position: Int) {

    companion object {

        const val INVALID_PAGE_ID: Long = -1

        const val TYPE_APPLIST_PAGE = 1
        const val TYPE_WIDGET_PAGE = 2
    }

}
